### DISCUSSION TITLE
[cloud_firestore] withConverter

### DISCUSSION THREAD

**Discussion by julienlebren** [firebase/flutterfire#6246] [Q&A] [answered]
The withConverter parameter is a good idea but I don't understand why we have to specify the required parameter "toFirestore" to make a query!? 
Why should I need to specify how the object should be converted **TO** Firestore if I'm just reading the data **FROM** Firestore?

Both fromFirestore and toFirestore should be optional in order to be used either when reading or updating the collection.
I don't see any case where it can be useful to use it both in the same time?

**Comment by rrousselGit** [comment, marked as answer]
Why would you _not_ specify both?

You should use the same reference both for reading and updating. You shouldn't use two separate refs for both reading and writing.

**Comment by julienlebren** [comment]
A simple case would be a view which is just displaying data without allowing the user to make any update.
Why would I specify **toFirestore** whereas it won't be used?

**Reply by rrousselGit** [reply]
You can throw:

```dart
toFirestore: (_, __) => throw UnimplementedError(),
```


Firestore doesn't know that you're not allowing writes. And if the parameter is not specified, the behaviour on write attempts is undefined

**Reply by julienlebren** [reply]
Yes, I'm already doing this, but it's a useless line of code.
That's why I was wondering why the parameters are required instead of being optional 😉
